### PR TITLE
[Feat]: Zoho OAuth Provider

### PR DESCRIPTION
**Pull request by UtkarshAhuja2003**
## What does this PR do?

Adds Zoho OAuth Provider

## Test Plan

Manual Testing

## Related PRs and Issues

- #7364 

## Checklist

- [x] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

## Screenshots
image

image

image

image

image

image

### CODE REVIEW

**Review by cbum-dev — COMMENTED** [review]
looks good..

File: src/Appwrite/Auth/OAuth2/Zoho.php [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+{
+    /**
+     * @var string
+     */
+    private string $endpoint = 'https://accounts.zoho.com';
+
+    /**
+     * @var string
+     */
+    private string $authEndpoint = 'https://accounts.zoho.com/oauth/v2/auth';
```
Comment (lohanidamodar) [inline src/Appwrite/Auth/OAuth2/Zoho.php:20]: This is used in only one place, can we remove this for now?

**Review by lohanidamodar — CHANGES_REQUESTED** [review]
Great work, can we address this, then we can merge

File: src/Appwrite/Auth/OAuth2/Zoho.php (hunk 1 above)
Comment (UtkarshAhuja2003) [reply src/Appwrite/Auth/OAuth2/Zoho.php]: Done

**Review by lohanidamodar — APPROVED** [review]

**Review by lohanidamodar — CHANGES_REQUESTED** [review]
@UtkarshAhuja2003  Thank you for quick update. Also, can you please fix the linter issues. Thank you

**Comment by UtkarshAhuja2003** [comment]
> @UtkarshAhuja2003 Thank you for quick update. Also, can you please fix the linter issues. Thank you

Done ✨

**Review by lohanidamodar — CHANGES_REQUESTED** [review]
@UtkarshAhuja2003 Thank you for your patience and I'm sorry to do this again, we need to get this PR to next version, can you please sync with the new base and resolve conflicts. 🙏🏻

**Comment by UtkarshAhuja2003** [comment]
> @UtkarshAhuja2003 Thank you for your patience and I'm sorry to do this again, we need to get this PR to next version, can you please sync with the new base and resolve conflicts. 🙏🏻

I've synced with the new base and resolved conflicts for the PR. Let me know if there's anything else needed. 🚀

**Review by eldadfux — APPROVED** [review]

**Comment by eldadfux** [comment]
Thank you for your contribution @UtkarshAhuja2003

**Pull request merged**